### ISSUE TITLE
Custom Trackers need to be re-added as a component after changes to the customHeader

### ISSUE THREAD

**Issue by Avdbergnmf** [immersivecognition/unity-experiment-framework#103]
Otherwise the Table headers aren't filled in the UXFDataTable.
```
InvalidOperationException: The row does not contain values for the same columns as the columns in the table!
Table: time
Row: right_pos_x, right_pos_y, right_pos_z, right_rot_x, right_rot_y, right_rot_z, left_pos_x, left_pos_y, left_pos_z, left_rot_x, left_rot_y, left_rot_z, time
UXF.UXFDataTable.AddCompleteRow (UXF.UXFDataRow newRow) (at Assets/UXF/Scripts/Etc/UXFDataTable.cs:109)
UXF.Tracker.RecordRow () (at Assets/UXF/Scripts/Etc/Tracker.cs:97)
UXF.Tracker.LateUpdate () (at Assets/UXF/Scripts/Etc/Tracker.cs:79)
``` 

Not sure if this is a bug or a feature, but it took me a good hour to figure out this problem.

Edit: 
Another solution could be to just mention this fact in the Wiki documentation!

**Comment by jackbrookes** [comment]
> and in adding values to the new UXFdatarow, the current timestamp looks to be added at the END of the array.

I think its not an ordering issue- the error message shows the `customHeader` would have been blank.


@Avdbergnmf can you explain what you actually did ("Custom Trackers need to be re-added as a component after changes to the customHeader"). When are you changing the header, and when are you re-adding the component?

**Comment by thefirstfloor** [comment]
In my case the situation was slightly different.

The `Position Rotation Tracker` script is on a custom prefab, but has all default UXF values in the header (so no `customheader`).
After adding it to the scene, disabling that component/script on the prefab in the hierarchy and adding a new one from the UXF folder seemed to fix the issue. Not very convenient but it works.

**Comment by Avdbergnmf** [comment]
> In my case the situation was slightly different.
> 
> The `Position Rotation Tracker` script is on a custom prefab, but has all default UXF values in the header (so no `customheader`). After adding it to the scene, disabling that component/script on the prefab in the hierarchy and adding a new one from the UXF folder seemed to fix the issue. Not very convenient but it works.

Glad u got it to work :)

**Comment by jackbrookes** [comment]
what are you guys trying to do by changing the custom values? they should be read-only:

![image](https://user-images.githubusercontent.com/23621596/154349776-c1c50fdf-835e-4b58-a499-9d29f11b4fd9.png)

**Comment by jackbrookes** [comment]
got it, I understand the issue, will fix! In the meantime delete/re-add. You may be able to right click -> reset the component.

**Comment by jackbrookes** [comment]
This is now fixed in UXF 2.4.1, with some improvements to the inspector. The issue is related to Unity Serialisation.
If you have a custom tracker script, you will need implement the two properties `CustomHeader` and `MeasurementDescriptor`.
See here: https://github.com/immersivecognition/unity-experiment-framework/wiki/Tracker-system
